polyfill/intersection-observer.js

```diff
@@ -267,9 +273,9 @@ IntersectionObserver.prototype._monitorIntersections = function() {
       addEvent(window, 'resize', this._checkForIntersections, true);
       addEvent(document, 'scroll', this._checkForIntersections, true);
 
-      if ('MutationObserver' in window) {
+      if (this.USE_MUTATION_OBSERVER && 'MutationObserver' in window) {
         this._domObserver = new MutationObserver(this._checkForIntersections);
-        this._domObserver.observe(document, {
+        this._domObserver.observe(this.root || document, {
```

Comment:
Sorry for the late comment (I was on vacation), but I'm not sure this should have been added. There are cases where mutations to an element outside of the root element could affect the positioning of one of the observed elements. So if we want to be safe, we should listen to all mutations.

Reply:
I understand the issue here. At first I though about a kind of resize sensor on the root element, but like you said, not only size but also positioning could be affected.
As users can now use the `USE_MUTATION_OBSERVER` to cut off unrelated spaming events to the watched element this one can be rolled back.